### ISSUE TITLE
Disabling "show autofill menu on form fields" in safari seems reset after every update.

### ISSUE THREAD

**Issue by jakeswenson** [bitwarden/clients#12170]
### Steps To Reproduce

1. In Safari, in the extension, goto settings > autofill
2. For "show autofill menu on form fields" select `Off`
3. Rejoice that this horrible feature is turned off
4. Update bitwarden 
5. Browse the web to something you want to login to
6. See the autofill box show up
7. Goto settings > autofill
8. See the selection set to when field is selected.
9. Cry yourself to sleep

### Expected Result

Selecting `Off` should be persisted permanently, across updates. I never want to use this feature.

### Actual Result

It seems every so often (after update??) this feature rises from the grave to haunt me. I think it knows I don't like it.

### Additional Context

This has happened to me for months, multiple times.

### Operating System

macOS

### Operating System Version

15.1.1 (24B2091)

### Web Browser

Safari

### Browser Version

Version 18.1.1 (20619.2.8.11.12)

### Build Version

2024.11.2

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by SergeantConfused** [comment]
Hello @jakeswenson,

Thank you for your report. This has been flagged internally to the Engineering department; please feel free to post additional information, such as screenshots or a screen video recordings, if you wish.

Thank you again,

**Comment by david-cako** [comment]
+1

This has been happening to me a lot this year.  Following updates through the Mac App Store, the Safari extension resets my "Show AutoFill on Form Fields" setting to on, when I prefer it off.  This may also happen with other settings.

---

#### Operating System

macOS

#### Operating System Version

15.1.1

#### Web Browser

Safari

#### Browser Version

Version 18.1.1 (20619.2.8.11.12)

#### Build Version

2024.11.2

**Comment by jakeswenson** [comment]
This just happened to reset on me again...  version still says: `Version: 2024.11.2`

**Comment by chimit** [comment]
The problem has returned:

```
Version: 2025.1.1
SDK: 'main (28c7e29)'
Server version: 2025.1.3
```

**Comment by toprowtech** [comment]
When updating from 2025.1.0 to 2025.1.1, the updater reset my selection for this setting:
Click items to autofill on Vault view

It was checked and then after updating to 2025.1.1, this setting was unchecked. Happened several times in different profiles, so it was definitely a setting that was reset by the new update.
Windows 11 24H2, Chrome.

PLEASE PLEASE PLEASE, stop changing user settings / preferences in updates. Updates should honor the existing/previous setting.

**Comment by speedyspikes** [comment]
This problem has been bugging me for a while.

**Comment by mmynk** [comment]
```
Version: 2025.2.2
SDK: 'main (6008e90)'
Server version: 2025.3.0
```

Issue still there, just trying to keep the thread alive.

**Comment by jakeswenson** [comment]
Yes this continues to happen, even now after the setting was changed from a dropdown to checkbox... I really don't understand why this it seams like _only setting continues to revert_.

**Comment by arnocs** [comment]
Same here,

System: Apple MacOS Sequoia 15.4.1 - Safari Version 18.4 (20621.1.15.11.10) - Bitwarden extention (AppStore) 2025.3.2
Issue: some preferences are not saved when bitwarden extention updates.

In my case / noticed, the "Autofill suggestions" are not saved after an extention update.

**Comment by audreyality** [comment]
Hey everybody, 

I'm looking into this bug. Our reproducibility steps are slightly different than what's reported here. We isolated the behavior to only requiring the plugin is enabled and disabled. That behavior has since been fixed (though I haven't isolated the release that resolved it).

Do any of you know if this is still happening on update? If so, then the bug we fixed was related, but insufficient to fix this bug. It would be very helpful to know if you installed Bitwarden using the .dmg or the App Store.


Otherwise, I'll be resolving this issue once we've verified that the enable/disable bug is fixed.

Thanks for your patience and help! ❤️

**Comment by jakeswenson** [comment]
> It would be very helpful to know if you installed Bitwarden using the .dmg or the App Store.

I Installed bitwarden via the app store. Which means updates were happening through that.

> We isolated the behavior to only requiring the plugin is enabled and disabled. 

Interesting. I wonder if this is because of how app store updates the app/extension? I certainly wasn't manually disabling/enabling. 

> Do any of you know if this is still happening on update?

I haven't consciously noticed it... so perhaps it has been fixed? I guess only time will tell, we can probably close and reopen as needed..

Thank you so much for finally tracking this down. Really appreciate it!

**Comment by audreyality** [comment]
It appears that this was resolved by an unrelated fix. Linked above. I'm closing this issue. Thanks for your help @jakeswenson!
